File: leetcode/easy/minimum-sum-of-four-digit-number-after-splitting-digits.py
# https://leetcode.com/problems/minimum-sum-of-four-digit-number-after-splitting-digits/

class Solution:
    def minimumSum(self, num: int) -> int:
        '''
        A = [0]*4
        for i in range(len(A)):
            A[i] = num % 10
            num = math.trunc(num / 10)
        A = sorted(A)
        return (A[0]*10+A[2]) + (A[1]*10+A[3])
        '''

        num = sorted(str(num), reverse=True)
        return int(num[0]) + int(num[1]) + int(num[2])*10 + int(num[3])*10

File: leetcode/easy/minimum-sum-of-four-digit-number-after-splitting-digits.ts
// https://leetcode.com/problems/minimum-sum-of-four-digit-number-after-splitting-digits/

// MINE
/*
function minimumSum(num: number): number {
    const A = num.toString().split('')
    A.sort((a, z) => parseInt(a) - parseInt(z))
    return parseInt(A[0]+A[2]) + parseInt(A[1]+A[3])
};
*/

// ANSWER: no string conversion
function minimumSum(num: number): number {
  const A = new Array(4).fill(0)
  for (let i = 0; i < 4; ++i) {
    A[i] = num % 10
    num = Math.trunc(num / 10)
  }
  A.sort((a, z) => a - z)
  return A[0] * 10 + A[2] + (A[1] * 10 + A[3])
}
